auto_submit/lib/requests/github_webhook.dart

```diff
@@ -17,7 +19,20 @@ class GithubWebhook {
   Future<Response> post(Request request) async {
     final Map<String, String> reqHeader = request.headers;
     logger.info('Header: $reqHeader');
+
+    // Listen to the pull request with 'autosubmit' label.
+    bool hasAutosubmit = false;
     final String rawBody = await request.readAsString();
+    final body = json.decode(rawBody) as Map<String, dynamic>;
+
+    if (body.containsKey('pull_request') && body['pull_request'].containsKey('labels')) {
```

Comment:
Instead of using the model, it's useful to create the list of that model:

```dart
List<IssueLabel> issues = body['pull_request']['labels'].map((dynamic label) => IssueLabel.fromJson(label)).toList();
```

A better approach would be:

```dart
PullRequest pullRequest = PullRequest.fromJson(body['pull_request']);
// pullRequest.labeles contains label.name == 'autosubmit'
```

Reply:
Hi Casey, if we get any of the request the body of which doesn't contain the 'pull_request', are we going to throw an error?

Reply:
Instead you should add a check and exit early in that case